web-server/assets/src/script/skills.js

```diff
@@ -65,6 +65,9 @@ const matchesFilters = (filters, string) => {
 }
 
 const filterSkillsList = filter => {
+  const queryString = filter === '' ? '' : `?f=${filter}`
+  history.replaceState({}, '', `skills${queryString}`)
```

Comment:
this is all `filterSkillsList` should do. There should be another method that is called to do the actual filtering. It needs to be called here, when the page loads and whenever the "history popstate" event is triggered.